test/tests/widgets/test_button.py

```diff
@@ -0,0 +1,22 @@
+from unittest.mock import Mock
+
+from pytest import fixture
+
+import toga
+
+from .common import test_background_color, test_color, test_text  # noqa: F401
+
+
+@fixture
+async def widget():
+    return toga.Button("")
+
+
+async def test_press(widget, probe):
```

Comment:
OK, let's discuss this tomorrow.

Reply:
After discussion, we came to the conclusion that:

* Yes, this is potentially confusing for people not particularly familiar with Pytest; however, it's also very much learnable knowledge, so anyone with moderate experience with Pytest should be familiar with the trick being used here
* In the specific case of these tests, we have a need for "a widget and a probe" for every test. That's likely to be a common property of almost all the individual widget tests - enough that having a single simple abstraction makes sense.
* If we have other multi-widget or complex-layout requirements, we can put them in separate testing packages with isolated conftest modules that won't inherit this specific widget + probe definitions.

Reply:
I think for someone who understands the basics of pytest, the only unusual thing going on here is that we have a `probe` fixture in a conftest file depending on a `widget` fixture in a test module. This is less common than the other direction, but conceptually it's no different from a method in a base class calling an abstract method which must be implemented by a subclass.

So maybe it would be clearer if the conftest file actually contained an abstract `widget` fixture which raises NotImplementedError, with a comment explaining how the test modules are supposed to override it. That way you could read the conftest file in isolation without being confused about where the `widget` was coming from. And if you forgot to define `widget` in a test module, then instead of pytest's generic "missing fixture" error, you would get a NotImplementedError leading you directly to the relevant information.

Anyway, I'll revisit this along with the comment above about the probe constructor.